doc/pages/development-guide/001-dev_guide.md

```diff
@@ -0,0 +1,165 @@
+# Development Guide
+
+Welcome to Astarte! To learn more about contributing to the [Astarte project](https://www.github.com/astarte-platform/astarte), check out the [Contributor's Guide](https://github.com/astarte-platform/astarte/blob/master/CONTRIBUTING.md).
+
+This guide is focused on writing and testing changes to Astarte. It assumes basic knowledge in the following areas:
+
+- You have read the [contributing guidelines](https://github.com/astarte-platform/astarte/blob/master/CONTRIBUTING.md) for the project
+- You are familiar with basic Astarte concepts such as Realms, Interfaces, Devices and Triggers. If you are not, you can learn more on the [documentation](001-intro_architecture.md)!
+- You have already run at least once [Astarte in 5 minutes](010-astarte_in_5_minutes.md)
+- If you plan to work on the code, you have some knowledge of Elixir and Docker
```

Comment:
since a setup from zero of elixir and erlang is non-trivial, (docker on the other hand is fairly easy to install), my suggestion is to write down somewhere our current suggested stack (asdf) and the relative plugins (troubleshooting also, for example like the ssl problem, maybe?)
If I remember correctly supported elixir versions also change from one astarte to another, that's another thing that I'd like to know if i started develop with astarte.

Nothing really blocking, as everybody could see the code, but nice to know imho

Reply:
+1 for a section for the environment setup. _maybe nix could be mentioned alongside asdf?_

Reply:
@eddbbt no need for that SSL troubleshooting section as Astarte 1.0 (i.e. using OTP < 25) is not supported anymore